cranelift/interpreter/src/interpreter.rs

```diff
@@ -203,6 +222,57 @@ impl<'a> InterpreterState<'a> {
         Self { functions, ..self }
     }
 
+    /// Registers a static heap and returns a reference to it
+    ///
+    /// This heap reference can be used to generate a heap pointer, which
+    /// can be used inside the interpreter to load / store values into the heap.
+    ///
+    /// ```rust
+    /// # use cranelift_codegen::ir::types::I64;
+    /// # use cranelift_interpreter::interpreter::{InterpreterState, HeapInit};
+    /// let mut state = InterpreterState::default();
+    /// let heap0 = state.register_heap(HeapInit::Zeroed(1024));
+    ///
+    /// let backing = Vec::from([10u8; 24]);
+    /// let heap1 = state.register_heap(HeapInit::FromBacking(backing));
+    /// ```
+    pub fn register_heap(&mut self, init: HeapInit) -> HeapId {
+        let heap_id = HeapId(self.heaps.len() as u32);
+
+        self.heaps.push(match init {
+            HeapInit::Zeroed(size) => iter::repeat(0).take(size).collect(),
+            HeapInit::FromBacking(backing) => backing,
```

Comment:
Do we need to check here that the length of the backing vec is at least as large as the size that was claimed in the `; heap: ...` directives, or somehow connect to that? Or is there an invariant that we will extend it if needed?

Reply:
This function here is somewhat independent of the testing infrastructure. And the interpreter doesn't really care about the size of the backing memory, since out of bounds writes/reads generate a trap.

The memory is allocated by the `RuntestEnvironment` struct which is the same struct that parses the `heap` directives. From there we just pass that memory directly into the interpreter, this happens in `test_interpret.rs` that's where we enforce that the heaps have the same size as the directives.